tests/scripts/test_answerstore.py

```diff
@@ -0,0 +1,314 @@
+    assert entries_created[0][1]['answer']['key2'] == 'False'
+    entries_created.pop()
+
+    # Testing scope1
+    assert store['scope1']['key1'] == 'scope1.key1'
+    assert store['scope1']['key2'] == 'scope1.key2'
+    result = a.get(scope='scope1', fallback='scope1')
+    assert result
+    assert result['key1'] == 'scope1.key1'
+    assert result['key2'] == 'scope1.key2'
+    assert entries_created
+    assert entries_created[0][1]['scope'] == 'scope1'
+    assert entries_created[0][1]['fallback'] == 'scope1'
+    assert entries_created[0][1]['answer']['key1'] == 'scope1.key1'
+    assert entries_created[0][1]['answer']['key2'] == 'scope1.key2'
+    entries_created.pop()
+
+    # Testing scope2
+    assert store['scope2']['key1'] == 'scope2.key1'
+    assert store['scope2']['key2'] == 'scope2.key2'
+
+    result = a.get(scope='scope2', fallback='scope2')
+    assert result
+    assert result['key1'] == 'scope2.key1'
+    assert result['key2'] == 'scope2.key2'
+    assert entries_created
+    assert entries_created[0][1]['scope'] == 'scope2'
+    assert entries_created[0][1]['fallback'] == 'scope2'
+    assert entries_created[0][1]['answer']['key1'] == 'scope2.key1'
+    assert entries_created[0][1]['answer']['key2'] == 'scope2.key2'
+    entries_created.pop()
+
+
+def test_answerfile_translate_for_workflow(answerfile):
+    m = MockManager()
+    a = AnswerStore(manager=m)
+    store = a._storage
+    a.load(answerfile)
+    assert store['boolscope']['key1'] == 'True'
+    assert store['boolscope']['key2'] == 'False'
+    assert store['scope1']['key1'] == 'scope1.key1'
+    assert store['scope1']['key2'] == 'scope1.key2'
+    assert store['scope2']['key1'] == 'scope2.key1'
+    assert store['scope2']['key2'] == 'scope2.key2'
+    a.translate_for_workflow(MockWorkflow)
+    assert store['boolscope']['key1'] is True
+    assert store['boolscope']['key2'] is False
+    assert store['scope1']['key1'] == 'scope1.key1'
+    assert store['scope1']['key2'] == 'scope1.key2'
+    assert store['scope2']['key1'] == 'scope2.key1'
+    assert store['scope2']['key2'] == 'scope2.key2'
+
+
+def test_answerfile_translate(answerfile):
+    m = MockManager()
+    a = AnswerStore(manager=m)
+    store = a._storage
+    a.load(answerfile)
+    assert store['boolscope']['key1'] == 'True'
+    assert store['boolscope']['key2'] == 'False'
+    assert store['scope1']['key1'] == 'scope1.key1'
+    assert store['scope1']['key2'] == 'scope1.key2'
+    assert store['scope2']['key1'] == 'scope2.key1'
+    assert store['scope2']['key2'] == 'scope2.key2'
+    a.translate(MockDialogBoolScope)
+    assert store['boolscope']['key1'] is True
+    assert store['boolscope']['key2'] is False
+    assert store['scope1']['key1'] == 'scope1.key1'
+    assert store['scope1']['key2'] == 'scope1.key2'
+    assert store['scope2']['key1'] == 'scope2.key1'
+    assert store['scope2']['key2'] == 'scope2.key2'
+
+
+def test_answerfile_generate(tmpdir, answerfile):
```

Comment:
some conditions of the tested function is not tested (i.e.  if hasattr(component, 'choices'):, if answer:)